host/Cargo.toml

```diff
@@ -23,6 +23,8 @@ static_cell = "2.1.0"
 # Logging
 log = { version = "0.4.16", optional = true }
 defmt = { version = "0.3", optional = true }
+num_enum = { version = "0.7.3", default-features = false }
+thiserror = { version = "2.0.0", default-features = false }
```

Comment:
@petekubiak Just to be sure, can you measure code size when enabling thiserror? I'm worried it adds some bloat.

Reply:
I checked out this commit which added `thiserror` (9f7bfc4) and ran `cargo size --release` on the esp32 example. I then did the same thing for the previous commit (000d1e1). Here are the results:
|| 000d1e1 | 9f7bfc4 |
|--------|--------|--------|
|text| 575266 | 575314 |
|data| 1144 | 1144 |
|bss| 144244 | 144244 |
|dec| 720654 | 720702 |
|hex| aff0e | aff3e | 

This would indicate that the inclusion of thiserror has only added 48 bytes, but I guess that only the code that's being used is being included in the binary.
Are there any other ways you'd suggest I should check this?

Reply:
No, that would be the same way I'd check it. Could you check with the nRF as well, which uses defmt. Increase in size when using log is expected. I'm a bit concerned if this adds to binary size if defmt is used.

Reply:
Running `cargo size --bin ble_bas_peripheral --release --features nrf52833` on the same two commits gives:
198820 bytes without `thiserror`
198840 bytes with `thiserror`

So only a change of 20 bytes in this case :)

Reply:
Ok, that is probably this line https://github.com/embassy-rs/trouble/blob/main/host/src/attribute.rs#L268  - which can be fixed separately.

Thanks for checking!